test/cql-pytest/test_permissions.py

```diff
@@ -271,40 +269,19 @@ def grant_idempotent():
                     for resource in [f'function {keyspace}.{fun}(int, list<int>)', f'all functions in keyspace {keyspace}', 'all functions']:
                         check_enforced(cql, username, permission='AUTHORIZE', resource=resource, function=grant_idempotent)
 
-# This test case is artificially extracted from the one above,
-# because it's qualified as cassandra_bug - the documentation quotes that ALTER is needed on
-# functions if the definition is replaced (CREATE OR REPLACE FUNCTION (...)),
-# and yet it's not enforced
-def test_grant_revoke_alter_udf_permissions(cassandra_bug, cql):
```

Comment:
I'm afraid I don't understand - this was earlier a separate test because you claimed (and explained in a comment) that you found a cassandra bug - something which work as the documentation says it should. No you removed this test - does it mean you no longer think this was a bug? You wrote something explaining this in the commit message, but I didn't understand the explanation.

Reply:
I wasn't actually the author of this test. I think this was marked a cassandra_bug due to a misunderstanding of the author, but the comment was convincing and I realized that this was actually not a bug only now with the help of the dtest that we recently started running. The misunderstanding was probably caused by the fact that you could run the `CREATE OR REPLACE` query with no ALTER permission. However, this was only the case if you were actually creating the UDF, and not replacing. On the contrary, if the UDF was actually existing before, you could run the `CREATE OR REPLACE` query without the CREATE permission, which can also sound a bit counter-intuitive. This would be also more difficult to implement, because you need to first check in the database whether the UDF already exists, and check the corresponding permission only afterwards.